pandas/sparse/array.py

```diff
@@ -503,6 +505,34 @@ def mean(self, axis=None, dtype=None, out=None):
             nsparse = self.sp_index.ngaps
             return (sp_sum + self.fill_value * nsparse) / (ct + nsparse)
 
+    def value_counts(self, dropna=True):
```

Comment:
Moved datetime-like logic to `algorithms.py` as it must handle `DatetimeTZBlock` also.